CMake compilation error (FindPackageMultipass)

I've been trying to compile a very recent version of `next` from https://github.com/boutproject/BOUT-dev/pull/2932 and got an error:
```

CMake Error at /ssd_scratch/BOUT-7d261d4/BOUT-dev/cmake/FindPackageMultipass.cmake:127 (check_c_source_compiles):
  Unknown CMake command "check_c_source_compiles".
Call Stack (most recent call first):
  /ssd_scratch/BOUT-7d261d4/BOUT-dev/cmake/FindPETSc.cmake:290 (multipass_source_compiles)
  /ssd_scratch/BOUT-7d261d4/BOUT-dev/cmake/FindPETSc.cmake:317 (petsc_test_compiles)
  /opt/york/easybuild/generic_arch/software/CMake/3.20.1-GCCcore-10.3.0/share/cmake-3.20/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /ssd_scratch/BOUT-7d261d4/BOUT-dev/build/bout++Config.cmake:138 (find_dependency)
  CMakeLists.txt:48 (find_package)


-- Configuring incomplete, errors occurred!
See also "/ssd_scratch/hermes-3/build-mc-upgrade-bout/CMakeFiles/CMakeOutput.log".
make: *** No targets specified and no makefile found. Stop.
```

Thanks to @ZedThree I was able to resolve it by changing `Runs` to `Compiles` in the below line:
https://github.com/boutproject/BOUT-dev/blob/c193e73fbeea206f42f69107395dcafc43f9c861/cmake/FindPackageMultipass.cmake#L111

I'm happy to commit the change if that's the right thing to do, but I don't understand CMake so can't judge that for myself.

--- a/cmake/FindPackageMultipass.cmake
+++ b/cmake/FindPackageMultipass.cmake
@@ -108,7 +108,7 @@ macro (MULTIPASS_C_SOURCE_RUNS includes libraries source runs)
 endmacro (MULTIPASS_C_SOURCE_RUNS)
 
 macro (MULTIPASS_SOURCE_COMPILES includes libraries source runs language)
-  include (Check${language}SourceRuns)
+  include (Check${language}SourceCompiles)
   # This is a ridiculous hack.  CHECK_${language}_SOURCE_* thinks that if the
   # *name* of the return variable doesn't change, then the test does
   # not need to be re-run.  We keep an internal count which we

